src/ai/backend/manager/registry.py

```diff
@@ -222,6 +223,25 @@
 SESSION_NAME_LEN_LIMIT = 10
 
 
+class SessionDestroyer(Protocol):
+    async def destroy_session(
+        self,
+        session: SessionRow,
+        *,
+        forced: bool = False,
+        reason: Optional[KernelLifecycleEventReason] = None,
+        user_role: UserRole | None = None,
+    ) -> Mapping[str, Any]:
+        """
+        Destroy session kernels. Do not destroy
+        CREATING/TERMINATING/ERROR and PULLING sessions.
+
+        :param forced: If True, destroy CREATING/TERMINATING/ERROR session.
+        :param reason: Reason to destroy a session if client wants to specify it manually.
+        """
+        ...
+
```

Comment:
Protocol should be defined where you use it.

Reply:
Thanks! This has been resolved with 7414f3ca2222feb980c1ad7ec8e04693ae93b9e1.